crates/iceberg/src/expr/predicate.rs

```diff
@@ -187,6 +199,14 @@ impl<T> SetExpression<T> {
         debug_assert!(op.is_set());
         Self { op, term, literals }
     }
+
+    pub(crate) fn op(&self) -> PredicateOperator {
+        self.op
+    }
+
+    pub(crate) fn literals(&self) -> &FnvHashSet<Datum> {
```

Comment:
```suggestion
    pub(crate) fn literals(&self) -> impl Iterator<Item=Datum> {
```

How about returning iterator only? It's better not expose the inner data structure.